staging/cse/windows/azurecnifunc.ps1

```diff
@@ -208,6 +210,23 @@ function Set-AzureCNIConfig
         }
     }
 
+    if ($IsIMDSRestrictionEnabled) {

```

Comment:
It'd be good to have some tests for this. There are 3 places I can think of:
* In the powershell unit (pester) tests. These will be in the same directory as this file.
* In the AgentBaker e2e tests. Look at scenario_win_test.go (I think it's called that). You should be able to create a new test that joins a node to a cluster, then run a powershell script on it to check that imds is blocked.
* In the aks-rp e2e tests, but that's outside the scope of this PR.

Reply:
Thanks! Added unit test via option 1, and I'll add corresponding E2E test in aks-rp repo once the change is available.